diff --git a/dist/index.js b/dist/index.js
index ea833a84c3323dcc72534e076d68473dfdb77128..41568478bea4246ff9acd7943bbd4d4ee6bd28e3 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -2096,8 +2096,8 @@ import { z as z12 } from "zod/v4";
 var xSearchArgsSchema = lazySchema4(
   () => zodSchema4(
     z12.object({
-      allowedXHandles: z12.array(z12.string()).max(10).optional(),
-      excludedXHandles: z12.array(z12.string()).max(10).optional(),
+      allowedXHandles: z12.array(z12.string()).max(20).optional(),
+      excludedXHandles: z12.array(z12.string()).max(20).optional(),
       fromDate: z12.string().optional(),
       toDate: z12.string().optional(),
       enableImageUnderstanding: z12.boolean().optional(),
diff --git a/src/tool/x-search.ts b/src/tool/x-search.ts
index 5566aa9c391f67cab60fe50ff2a2b55ca71b0b73..6d0dd586f07c5ea9d2bd2dcbed4be211be2e3c8a 100644
--- a/src/tool/x-search.ts
+++ b/src/tool/x-search.ts
@@ -8,8 +8,8 @@ import { z } from 'zod/v4';
 export const xSearchArgsSchema = lazySchema(() =>
   zodSchema(
     z.object({
-      allowedXHandles: z.array(z.string()).max(10).optional(),
-      excludedXHandles: z.array(z.string()).max(10).optional(),
+      allowedXHandles: z.array(z.string()).max(20).optional(),
+      excludedXHandles: z.array(z.string()).max(20).optional(),
       fromDate: z.string().optional(),
       toDate: z.string().optional(),
       enableImageUnderstanding: z.boolean().optional(),
